pubspec.yaml

```diff
@@ -1,5 +1,5 @@
 name: rudder_sdk_flutter
-version: 1.0.7
+version: 1.0.8
```

Comment:
I would advise against using 1.0.8 because this PR contains breaking changes in the interface such as setAnonymousId -> putAnonymousId

Either the old interface is market as deprecated and it calls the new functions, or it should be a major version, otherwise all the builds of people using this package will break.

Reply:
> 

This may be true, but 1.0.7 is completely broken for flutter [Android] apps at the moment, due to [this commit](https://github.com/rudderlabs/rudder-sdk-android/commit/8bde3a27d32ede0387088cebde01aa340edfae6a). 

```
/home/jeff/.pub-cache/hosted/pub.dartlang.org/rudder_sdk_flutter-1.0.7/android/src/main/java/com/rudderstack/sdk/flutter/RudderSdkFlutterPlugin.java:211: error: cannot find symbol
                RudderClient.updateWithAdvertisingId(
                            ^
  symbol:   method updateWithAdvertisingId(String)
  location: class RudderClient
/home/jeff/.pub-cache/hosted/pub.dartlang.org/rudder_sdk_flutter-1.0.7/android/src/main/java/com/rudderstack/sdk/flutter/RudderSdkFlutterPlugin.java:221: error: cannot find symbol
                RudderClient.setAnonymousId((String) argumentsMap.get("anonymousId"));
                            ^
  symbol:   method setAnonymousId(String)
  location: class RudderClient
```

Reply:
IMHO this is because also the android SDK changes didn't follow semver.

I would keep the same Dart interface available, mark it as deprecated and then if the android SDK does not follow semver I'd advise pinning the android SDK dependency to a specific version as well.

Doing this kind of breaking changes in minor version updates means that CI/CD pipelines suddendly break and it doesn't encourage clients to use and rely on Rudderstack.

Reply:
Completely agree -- what you're describing is exactly what happened to us. I will open a separate ticket specific to this issue.

Reply:
Hey @savy-91 & @corcoran,

We completely agree and understand you. 

We are about to change it to a major version release and also we will pin the specific version of Android SDK. 

We will also mark the previous API's as deprecated. Thank you guys for pointing it out

Reply:
@savy-91 @corcoran,

We are marking the earlier Api's as deprecated and forwarding the calls from them to the latest API's which we had brought in to have the same naming convention across all the Mobile SDK's.

We had planned to remove them completely in a Major release later, and for now, we will be making a minor release.

Also, we will not be making any breaking changes in our Native SDK's as well and we marked the earlier API's as deprecated and forwarded them to the latest Api's and released them.

We will be adopting Semver strictly from now. 

Thank you for the support.